sympy/concrete/products.py

```diff
@@ -399,6 +399,21 @@ def _eval_product_direct(self, term, limits):
         (k, a, n) = limits
         return Mul(*[term.subs(k, a + i) for i in range(n - a + 1)])
 
+    def _eval_derivative(self, x):
+        from sympy.concrete.summations import Sum, Dummy, Symbol, Derivative
```

Comment:
It's probably best to import Dummy, Symbol and Derivative at the top of the file like sympy/concrete/summations.py does.

Reply:
For importing `Sum` at the top, it raises `ImportError` maybe due to circular imports as `Sum` has been imported several times inside many methods, even on removing those imports and defining at the top still raises the error. For `Symbol` and `Derivative`, it works fine. 
Should I try to fix it, or let `Sum` be defined in the method only?

Reply:
Yeah, please put them at the top and keep `Sum` in the method. If it's possible, please also put `Dummy` at the top. I suggest to import them the same way as syummations.py does.

Reply:
Thank you for replying, kindly review the latest commit.
Thanks.